Resources/doc/first_steps.rst

```diff
@@ -0,0 +1,456 @@
+in a number of different formats including XML or directly inside the
+``Product`` class via annotations:
+
+.. configuration-block::
+
+    .. code-block:: php
+
+        // src/Acme/StoreBundle/Document/Product.php
+        namespace Acme\StoreBundle\Document;
+
+        use Doctrine\ODM\MongoDB\Mapping\Annotations as MongoDB;
+
+        /**
+         * @MongoDB\Document
+         */
+        class Product
+        {
+            /**
+             * @MongoDB\Id
+             */
+            protected $id;
+
+            /**
+             * @MongoDB\Field(type="string")
+             */
+            protected $name;
+
+            /**
+             * @MongoDB\Field(type="float")
+             */
+            protected $price;
+        }
+
+    .. code-block:: xml
+
+        <!-- src/Acme/StoreBundle/Resources/config/doctrine/Product.mongodb.xml -->
+        <doctrine-mongo-mapping xmlns="http://doctrine-project.org/schemas/odm/doctrine-mongo-mapping"
+              xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+              xsi:schemaLocation="http://doctrine-project.org/schemas/odm/doctrine-mongo-mapping
+                            http://doctrine-project.org/schemas/odm/doctrine-mongo-mapping.xsd">
+
+            <document name="Acme\StoreBundle\Document\Product">
+                <field fieldName="id" id="true" />
+                <field fieldName="name" type="string" />
+                <field fieldName="price" type="float" />
+            </document>
+        </doctrine-mongo-mapping>
+
+Doctrine allows you to choose from a wide variety of different field types,
+each with their own options. For information on the available field types,
+see the :ref:`cookbook-mongodb-field-types` section.
+
+.. seealso::
+
+    You can also check out Doctrine's `Basic Mapping Documentation`_ for
+    all details about mapping information. If you use annotations, you'll
+    need to prepend all annotations with ``MongoDB\`` (e.g. ``MongoDB\String``),
+    which is not shown in Doctrine's documentation. You'll also need to include
+    the ``use Doctrine\ODM\MongoDB\Mapping\Annotations as MongoDB;`` statement,
+    which *imports* the ``MongoDB`` annotations prefix.
+
+Persisting Objects to MongoDB
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Now that you have a mapped ``Product`` document complete with getter and
+setter methods, you're ready to persist data to MongoDB. From inside a controller,
+this is pretty easy. Add the following method to the ``DefaultController``
+of the bundle:
+
+.. code-block:: php
+    :linenos:
+
+    // src/Acme/StoreBundle/Controller/DefaultController.php
+    use Acme\StoreBundle\Document\Product;
+    use Symfony\Component\HttpFoundation\Response;
+    // ...
+
+    public function createAction()
+    {
+        $product = new Product();
+        $product->setName('A Foo Bar');
+        $product->setPrice('19.99');
+
+        $dm = $this->get('doctrine_mongodb')->getManager();

```

Comment:
This should be changed to dependency injection.

Reply:
Same as previous comment, I'll be fixing these separately